py-polars/polars/expr/expr.py

```diff
@@ -527,41 +527,228 @@ def alias(self, name: str) -> Self:
 
         Examples
         --------
+        Rename an expression to avoid overwriting an existing column.
+
         >>> df = pl.DataFrame(
         ...     {
         ...         "a": [1, 2, 3],
-        ...         "b": ["a", "b", None],
+        ...         "b": ["x", "y", "z"],
         ...     }
         ... )
-        >>> df
-        shape: (3, 2)
-        ┌─────┬──────┐
-        │ a   ┆ b    │
-        │ --- ┆ ---  │
-        │ i64 ┆ str  │
-        ╞═════╪══════╡
-        │ 1   ┆ a    │
-        │ 2   ┆ b    │
-        │ 3   ┆ null │
-        └─────┴──────┘
-        >>> df.select(
-        ...     pl.col("a").alias("bar"),
-        ...     pl.col("b").alias("foo"),
+        >>> df.with_columns(
+        ...     pl.col("a") + 10,
+        ...     pl.col("b").str.to_uppercase().alias("c"),
         ... )
-        shape: (3, 2)
-        ┌─────┬──────┐
-        │ bar ┆ foo  │
-        │ --- ┆ ---  │
-        │ i64 ┆ str  │
-        ╞═════╪══════╡
-        │ 1   ┆ a    │
-        │ 2   ┆ b    │
-        │ 3   ┆ null │
-        └─────┴──────┘
+        shape: (3, 3)
+        ┌─────┬─────┬─────┐
+        │ a   ┆ b   ┆ c   │
+        │ --- ┆ --- ┆ --- │
+        │ i64 ┆ str ┆ str │
+        ╞═════╪═════╪═════╡
+        │ 11  ┆ x   ┆ X   │
+        │ 12  ┆ y   ┆ Y   │
+        │ 13  ┆ z   ┆ Z   │
+        └─────┴─────┴─────┘
+
+        Overwrite the default name of literal columns to prevent errors for duplicate
```

Comment:
* _with_ duplicate column names :)

Reply:
Uh that sounds weird to me - maybe we do: errors "because of" duplicate column names?